src/lando/middleware.py

```diff
@@ -0,0 +1,41 @@
+from django.conf import settings
+from django.http import HttpRequest, HttpResponse
+
+
+class ResponseHeadersMiddleware:
+    """Add custom response headers for each request."""
+
+    def __init__(self, get_response: callable):
+        self.get_response = get_response
+
+    def __call__(self, request: HttpRequest) -> HttpResponse:
+        # NOTE: These headers were ported from both the legacy UI and API.
+        # API specific CSP headers should be implemented as part of bug 1927163.
+
+        response = self.get_response(request)
+
+        response["X-Frame-Options"] = "DENY"
+        response["X-Content-Type-Options"] = "nosniff"
+
+        csp = [
+            "default-src 'self'",
+            "font-src 'self' https://code.cdn.mozilla.net",
+            "style-src 'self' https://code.cdn.mozilla.net",
+            "img-src 'self' *.cloudfront.net *.gravatar.com *.googleusercontent.com",
+            "object-src 'none'",
+            "frame-ancestors 'none'",
+            "manifest-src 'none'",
+            "worker-src 'none'",
+            "media-src 'none'",
+            "frame-src 'none'",
+            "base-uri 'none'",
```

Comment:
Nit: we could sort these for readability.

Reply:
I considered that, however, there is an implication in the specs that the order is preserved (although it is not respected), however that is not explicitly stated in the specs, so I decided to keep those as they appeared in the [original code](https://github.com/mozilla-conduit/lando-ui/blob/main/landoui/app.py#L63).

Reply:
Works for me. :)